github/table_github_code_owner.go

```diff
@@ -0,0 +1,157 @@
+package github
+
+import (
+	"context"
+	"github.com/google/go-github/v45/github"
+	"strings"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+)
+
+//// TABLE DEFINITION
+
+func gitHubCodeOwnerColumns() []*plugin.Column {
+	return []*plugin.Column{
+		// Top columns
+		{Name: "repository_full_name", Type: proto.ColumnType_STRING, Description: "The full name of the repository, including the owner and repo name."},
+		// Other columns
+		{Name: "pattern", Type: proto.ColumnType_STRING, Description: "The pattern used to identify what code a team, or an individual is responsible for"},
+		{Name: "users", Type: proto.ColumnType_JSON, Description: "Users responsible for code in the repo"},
+		{Name: "teams", Type: proto.ColumnType_JSON, Description: "Teams responsible for code in the repo"},
+		{Name: "pre_comments", Type: proto.ColumnType_JSON, Description: "Specifies the comments added above a key."},
+		{Name: "line_comment", Type: proto.ColumnType_STRING, Description: "Specifies the comment following the node and before empty lines."},
+	}
+}
+
+//// TABLE DEFINITION
+
+func tableGithubCodeOwner() *plugin.Table {
+	return &plugin.Table{
+		Name:        "github_code_owner",
+		Description: "Individuals or teams that are responsible for code in a repository.",
+		List: &plugin.ListConfig{
+			Hydrate:           tableGitHubCodeOwnerList,
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			KeyColumns:        plugin.SingleColumn("repository_full_name"),
+		},
+		Columns: gitHubCodeOwnerColumns(),
+	}
+}
+
+// // LIST FUNCTION
+type CodeOwnerRule struct {
+	Pattern     string
+	Users       []string
+	Teams       []string
+	PreComments []string
+	LineComment string
+}
+
+func tableGitHubCodeOwnerList(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("tableGitHubCodeOwnerList")
+	repoFullName := d.KeyColumnQuals["repository_full_name"].GetStringValue()
+	owner, repoName := parseRepoFullName(repoFullName)
+
+	type CodeOwnerRuleResponse struct {
+		RepositoryFullName string
+		Pattern            string
+		Users              []string
+		Teams              []string
+		PreComments        []string
+		LineComment        string
+	}
+	getCodeOwners := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+		var fileContent *github.RepositoryContent
+		var err error
+
+		client := connect(ctx, d)
+		opt := &github.RepositoryContentGetOptions{}
+		// stop on the first found CODEOWNERS file
+		var paths = []string{"CODEOWNERS", ".github/CODEOWNERS", "docs/CODEOWNERS"}
```

Comment:
Can a repo have multiple CODEOWNERS files (even in GitHub throws up warnings in the UI)? If so, which one is used by the table (looks like it's the order defined in the array)? Is there a precedence that GitHub has on which CODEOWNERS file it will use?

Reply:
I did some tests on a repo : there is no composition between files (some rules from a file, and others from another) and there is a precedence :
- `.github/CODEOWNERS`
- `CODEOWNERS`
- `docs/CODEOWNERS`

I changed my code to respect this order.

Reply:
Thanks for checking @aminvielledebatAtBedrock , can you please also include a comment that has a link where it specifies this for future reference?

Reply:
Comment added ! 
Thanks for the review !